brainframe/api/bf_codecs/alarm_codecs.py

```diff
@@ -69,45 +70,44 @@ def from_dict(d):
                          stream_id=d["stream_id"])
 
 
+@dataclass
 class Alert(Codec):
     """This is sent when an Alarm has been triggered."""
 
-    def __init__(self, *,
-                 alarm_id: int,
-                 zone_id: int,
-                 stream_id: int,
-                 start_time: float,
-                 end_time: float,
-                 verified_as: Optional[bool],
-                 id_: int = None):
-        self.id: int = id_
-        """A unique identifier"""
-        self.alarm_id: int = alarm_id
-        """The ID of the alarm that this alert came from"""
-        self.zone_id: int = zone_id
-        """The ID of the zone this alert pertains to"""
-        self.stream_id: int = stream_id
-        """The ID of the stream this alert pertains to"""
-        self.start_time: float = start_time
-        """When the event started happening, in Unix Time (seconds)"""
-        self.end_time: Optional[float] = end_time
-        """When the event stopped happening, in Unix Time (seconds), or None
-        if the alert is ongoing.
-        """
-        self.verified_as: Optional[bool] = verified_as
-        """
-        - If True, then this alert was labeled by a person as legitimate
-        - If False, then this alert was labeled by a person as a false alarm
-        - If None, then this alert has not been reviewed by a person
-        """
+    alarm_id: int
+    """The ID of the alarm that this alert came from"""
+
+    zone_id: int
+    """The ID of the zone this alert pertains to"""
+
+    stream_id: int
+    """The ID of the stream this alert pertains to"""
+
+    start_time: float
+    """When the event started happening, in Unix Time (seconds)"""
+
+    end_time: Optional[float]
+    """When the event stopped happening, in Unix Time (seconds), or None
+    if the alert is ongoing.
+    """
+
+    verified_as: Optional[bool]
+    """
+    - If True, then this alert was labeled by a person as legitimate
+    - If False, then this alert was labeled by a person as a false alarm
+    - If None, then this alert has not been reviewed by a person
+    """
+
+    id: Optional[int] = None
+    """A unique identifier"""
 
     def to_dict(self):
```

Comment:
can we inherit from a class/use a metaclass that provides the to_dict method?

Reply:
We'll pick this back up in the dataclasses PR but I think we could do something here.

Reply:
If that's cool with you